### PR TITLE
Upgrade controller-gen, operator-sdk, kubebuilder and Go versions

### PR DESCRIPTION
**Pull request by levan-m**
### What does this PR do?

Upgrades CRD generation dependencies and Go to 1.22. 

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Deploy locally and used minimal DDA to sanity check DDA status updates.

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/configuration.v2alpha1.md [hunk 1]
```diff
@@ -33,190 +33,190 @@ spec:
...
-| features.admissionController.agentSidecarInjection.image.pullPolicy | The Kubernetes pull policy: Use Always, Never, or IfNotPresent. |
-| features.admissionController.agentSidecarInjection.image.pullSecrets | It is possible to specify Docker registry credentials. See https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod |
-| features.admissionController.agentSidecarInjection.image.tag | Define the image tag to use. To be used if the Name field does not correspond to a full image string. |
+| features.admissionController.agentCommunicationMode | AgentCommunicationMode corresponds to the mode used by the Datadog application libraries to communicate with the Agent.It can be "hostip", "service", or "socket". |
```
Comment (fanny-jiang) [inline docs/configuration.v2alpha1.md:36]: This update is removing the spaces between sentences

File: docs/configuration.v2alpha1.md [hunk 2]
```diff
@@ -249,123 +249,123 @@ In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.
...
-| [key].containers.[key].livenessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1. |
+| [key].containers.[key].env `[]object` | Specify additional environment variables in the container.See also: https://docs.datadoghq.com/agent/kubernetes/?tab=helm#environment-variables |
+| [key].containers.[key].healthPort | HealthPort of the container for the internal liveness probe.Must be the same as the Liveness/Readiness probes. |
+| [key].containers.[key].livenessProbe.exec.command | Command is the command line to execute inside the container, the working directory for thecommand  is root ('/') in the container's filesystem. The command is simply exec'd, it isnot run inside a shell, so traditional shell instructions ('|', etc) won't work. To usea shell, you need to explicitly call out to that shell.Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:265]: It seems like the spaces shouldn't have been removed here?

File: docs/configuration.v2alpha1.md [hunk 3]
```diff
@@ -249,123 +249,123 @@ In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.
...
+| [key].containers.[key].livenessProbe.tcpSocket.port | Number or name of the port to access on the container.Number must be in the range 1 to 65535.Name must be an IANA_SVC_NAME. |
+| [key].containers.[key].livenessProbe.terminationGracePeriodSeconds | Optional duration in seconds the pod needs to terminate gracefully upon probe failure.The grace period is the duration in seconds after the processes running in the pod are senta termination signal and the time when the processes are forcibly halted with a kill signal.Set this value longer than the expected cleanup time for your process.If this value is nil, the pod's terminationGracePeriodSeconds will be used. Otherwise, thisvalue overrides the value provided by the pod spec.Value must be non-negative integer. The value zero indicates stop immediately viathe kill signal (no opportunity to shut down).This is a beta field and requires enabling ProbeTerminationGracePeriod feature gate.Minimum value is 1. spec.terminationGracePeriodSeconds is used if unset. |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:279]: And here

File: docs/configuration.v2alpha1.md [hunk 4]
```diff
@@ -249,123 +249,123 @@ In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.
...
 | [key].containers.[key].name | Name of the container that is overridden |
-| [key].containers.[key].readinessProbe.exec.command | Command is the command line to execute inside the container, the working directory for the command  is root ('/') in the container's filesystem. The command is simply exec'd, it is not run inside a shell, so traditional shell instructions ('|', etc) won't work. To use a shell, you need to explicitly call out to that shell. Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |
-| [key].containers.[key].readinessProbe.failureThreshold | Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1. |
+| [key].containers.[key].readinessProbe.exec.command | Command is the command line to execute inside the container, the working directory for thecommand  is root ('/') in the container's filesystem. The command is simply exec'd, it isnot run inside a shell, so traditional shell instructions ('|', etc) won't work. To usea shell, you need to explicitly call out to that shell.Exit status of 0 is treated as live/healthy and non-zero is unhealthy. |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:283]: Accident?

**Review by rtrieu — CHANGES_REQUESTED** [review]
In many places within `configuration.v2alpha.md`, it seems like there were spaces that were removed by accident.

File: docs/configuration.v2alpha1.md (hunk 2 above)
Comment (levan-m) [reply docs/configuration.v2alpha1.md]: Thanks! This should be fixed by 5th commit.

**Review by khewonc — APPROVED** [review]

File: docs/configuration.v2alpha1.md [hunk 5]
```diff
@@ -33,190 +33,190 @@ spec:
...
-| features.cws.syscallMonitorEnabled | SyscallMonitorEnabled enables Syscall Monitoring (recommended for troubleshooting only).Default: false |
-| features.dogstatsd.hostPortConfig.enabled | Enabled enables host port configurationDefault: false |
-| features.dogstatsd.hostPortConfig.hostPort | Port takes a port number (0 < x < 65536) to expose on the host. (Most containers do not need this.)If HostNetwork is enabled, this value must match the ContainerPort. |
+| features.cws.enabled | Enabled enables Cloud Workload Security. Default: false |
+| features.cws.network.enabled | Enabled enables Cloud Workload Security Network detections. Default: true |
+| features.cws.remoteConfiguration.enabled | Enabled enables Remote Configuration for Cloud Workload Security. Default: true |
+| features.cws.securityProfiles.enabled | Enabled enables Security Profiles collection for Cloud Workload Security. Default: true |
+| features.cws.syscallMonitorEnabled | SyscallMonitorEnabled enables Syscall Monitoring (recommended for troubleshooting only). Default: false |
+| features.dogstatsd.hostPortConfig.enabled | Enabled enables host port configuration Default: false |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:84]:
```suggestion
| features.dogstatsd.hostPortConfig.enabled | Enabled enables host port configuration. Default: false |
```

File: docs/configuration.v2alpha1.md [hunk 6]
```diff
@@ -33,190 +33,190 @@ spec:
...
-| features.externalMetricsServer.useDatadogMetrics | UseDatadogMetrics enables usage of the DatadogMetrics CRD (allowing one to scale on arbitrary Datadog metric queries).Default: true |
-| features.externalMetricsServer.wpaController | WPAController enables the informer and controller of the Watermark Pod Autoscaler.NOTE: The Watermark Pod Autoscaler controller needs to be installed.See also: https://github.com/DataDog/watermarkpodautoscaler.Default: false |
-| features.helmCheck.collectEvents | CollectEvents set to `true` enables event collection in the Helm check(Requires Agent 7.36.0+ and Cluster Agent 1.20.0+)Default: false |
-| features.helmCheck.enabled | Enabled enables the Helm check.Default: false |
-| features.helmCheck.valuesAsTags | ValuesAsTags collects Helm values from a release and uses them as tags(Requires Agent and Cluster Agent 7.40.0+).Default: {} |
+| features.externalMetricsServer.port | Port specifies the metricsProvider External Metrics Server service port. Default: 8443 |
+| features.externalMetricsServer.registerAPIService | RegisterAPIService registers the External Metrics endpoint as an APIService Default: true |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:104]:
```suggestion
| features.externalMetricsServer.registerAPIService | RegisterAPIService registers the External Metrics endpoint as an APIService. Default: true |
```

File: docs/configuration.v2alpha1.md [hunk 7]
```diff
@@ -33,190 +33,190 @@ spec:
...
+| features.kubeStateMetricsCore.enabled | Enabled enables Kube State Metrics Core. Default: true |
+| features.liveContainerCollection.enabled | Enables container collection for the Live Container View. Default: true |
+| features.liveProcessCollection.enabled | Enabled enables Process monitoring. Default: false |
+| features.liveProcessCollection.scrubProcessArguments | ScrubProcessArguments enables scrubbing of sensitive data in process command-lines (passwords, tokens, etc. ). Default: true |
+| features.liveProcessCollection.stripProcessArguments | StripProcessArguments enables stripping of all process arguments. Default: false |
+| features.logCollection.containerCollectAll | ContainerCollectAll enables Log collection from all containers. Default: false |
+| features.logCollection.containerCollectUsingFiles | ContainerCollectUsingFiles enables log collection from files in `/var/log/pods instead` of using the container runtime API. Collecting logs from files is usually the most efficient way of collecting logs. See also: https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#log-collection-setup Default: true |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:119]:
```suggestion
| features.logCollection.containerCollectUsingFiles | ContainerCollectUsingFiles enables log collection from files in `/var/log/pods instead` of using the container runtime API. Collecting logs from files is usually the most efficient way of collecting logs. See also: https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#log-collection-setup. Default: true |
```

File: docs/configuration.v2alpha1.md [hunk 8]
```diff
@@ -33,190 +33,190 @@ spec:
...
+| features.logCollection.containerLogsPath | ContainerLogsPath allows log collection from the container log path. Set to a different path if you are not using the Docker runtime. See also: https://docs.datadoghq.com/agent/kubernetes/daemonset_setup/?tab=k8sfile#create-manifest Default: `/var/lib/docker/containers` |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:120]:
```suggestion
| features.logCollection.containerLogsPath | ContainerLogsPath allows log collection from the container log path. Set to a different path if you are not using the Docker runtime. See also: https://docs.datadoghq.com/agent/kubernetes/daemonset_setup/?tab=k8sfile#create-manifest. Default: `/var/lib/docker/containers` |
```

File: docs/configuration.v2alpha1.md [hunk 9]
```diff
@@ -33,190 +33,190 @@ spec:
...
+| features.logCollection.containerSymlinksPath | ContainerSymlinksPath allows log collection to use symbolic links in this directory to validate container ID -> pod. Default: `/var/log/containers` |
+| features.logCollection.enabled | Enabled enables Log collection. Default: false |
+| features.logCollection.openFilesLimit | OpenFilesLimit sets the maximum number of log files that the Datadog Agent tails. Increasing this limit can increase resource consumption of the Agent. See also: https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#log-collection-setup Default: 100 |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:123]:
```suggestion
| features.logCollection.openFilesLimit | OpenFilesLimit sets the maximum number of log files that the Datadog Agent tails. Increasing this limit can increase resource consumption of the Agent. See also: https://docs.datadoghq.com/agent/basic_agent_usage/kubernetes/#log-collection-setup. Default: 100 |
```

File: docs/configuration.v2alpha1.md [hunk 10]
```diff
@@ -33,190 +33,190 @@ spec:
...
+| features.logCollection.podLogsPath | PodLogsPath allows log collection from a pod log path. Default: `/var/log/pods` |
+| features.logCollection.tempStoragePath | TempStoragePath (always mounted from the host) is used by the Agent to store information about processed log files. If the Agent is restarted, it starts tailing the log files immediately. Default: `/var/lib/datadog-agent/logs` |
+| features.npm.collectDNSStats | CollectDNSStats enables DNS stat collection. Default: false |
+| features.npm.enableConntrack | EnableConntrack enables the system-probe agent to connect to the netlink/conntrack subsystem to add NAT information to connection data. See also: http://conntrack-tools.netfilter.org/ Default: false |
```
Comment (rtrieu) [inline docs/configuration.v2alpha1.md:127]:
```suggestion
| features.npm.enableConntrack | EnableConntrack enables the system-probe agent to connect to the netlink/conntrack subsystem to add NAT information to connection data. See also: http://conntrack-tools.netfilter.org/. Default: false |
```

**Review by rtrieu — APPROVED** [review]
Found a lot of spacing / period errors, but approving!

**Comment by levan-m** [comment]
These docs are getting generated from existing Go file and weren't caused by this PR so I'll merge it as is.

**Pull request merged**